### PR TITLE
fixes #314

### LINKED ISSUE #314: Misp feed fails
Before opening an issue, please make do a quick search through the current issues
and the mailing list (https://groups.google.com/forum/#!forum/yeti-users) to see
if it hasn't been answered already.

### Description
MISP feed integration fail

### Environment

Ubuntu 18.04

### Steps to Reproduce

python testfeeds.py Misp-Feed

#### Expected behavior
MISP attributes on YETI

#### Actual behavior

root@host:~/yeti# python testfeeds.py Misp-Feed
DEBUG:root:Scheduler started
Running Misp-Feed...
DEBUG:root:Running Misp-Feed (ID: 5bf354872eec1859bdfca5e5)
DEBUG:root:Processing instance misp_1
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): misp.example.com:443
DEBUG:urllib3.connectionpool:https://misp.example.com:443 "GET /organisations/index/scope:all HTTP/1.1" 200 None
DEBUG:root:Getting all events for misp_1
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): misp.exame.com:443
DEBUG:urllib3.connectionpool:https://misp.example.com:443 "POST /events/restSearch HTTP/1.1" 200 None
ERROR:root:Traceback (most recent call last):
  File "/root/yeti/core/feed.py", line 42, in update_feed
    f.update()
  File "/root/yeti/plugins/feeds/public/misp.py", line 151, in update
    self.get_all_events(instance)
  File "/root/yeti/plugins/feeds/public/misp.py", line 131, in get_all_events
    for date_from, date_to, imported in self.week_events(instance):
  File "/root/yeti/plugins/feeds/public/misp.py", line 104, in week_events
    self.analyze(event['Event'], instance)
  File "/root/yeti/plugins/feeds/public/misp.py", line 177, in analyze
    galaxies = self.instances[instance]['galaxy_filter'].split(',')
AttributeError: 'NoneType' object has no attribute 'split'

ERROR:root:ERROR updating feed: 'NoneType' object has no attribute 'split'

### PR DESCRIPTION
**Pull request by sebdraven**

### CODE REVIEW

File: plugins/feeds/public/misp.py [hunk 1]
```diff
@@ -171,7 +171,8 @@ def analyze(self, event, instance):
         if external_analysis:
             context['external sources'] = '\r\n'.join(external_analysis)
 
-        if 'galaxy_filter' not in self.instances[instance]:
+        if 'galaxy_filter' not in self.instances[instance] \
```
Comment (p-l-) [inline plugins/feeds/public/misp.py]: I would suggest something easier to read & understand, like:
```Python
         if not self.instances[instance].get('galaxy_filter'):
```

**Comment by sebdraven** [comment]
Yep good remark,
I do not usually use this function. 

Thanks

File: plugins/feeds/public/misp.py (hunk 1 above)
Comment (sebdraven) [reply plugins/feeds/public/misp.py]: I do not usually use this function

Comment (sebdraven) [reply plugins/feeds/public/misp.py]: Yep good remark,
I do not usually use this function.

Thanks

**Comment by p-l-** [comment]
While you are changing this, you might want to inverse the test, to make the code even more readable:
```Python
        if self.instances[instance].get('galaxy_filter'):
            galaxies = self.instances[instance]['galaxy_filter'].split(',')
        else:
            tags = [tag['name'] for tag in event['Tag']]
```

Or, IMO even better:
```Python
        try:
            galaxies = self.instances[instance]['galaxy_filter'].split(',')
        except (KeyError, AttributeError):
            tags = [tag['name'] for tag in event['Tag']]
```

**Pull request merged**